src/Data/List/Ext.agda

```diff
@@ -1,12 +1,51 @@
 {-# OPTIONS --safe #-}
 module Data.List.Ext where
 
-open import Agda.Builtin.List using (List; []; _∷_)
-open import Agda.Builtin.Maybe using (Maybe; nothing; just)
-open import Agda.Builtin.Nat using (Nat; zero; suc)
+open import Data.List using (List; [_]; _++_; map; head; drop; concatMap)
+open import Data.List.Membership.Propositional using (_∈_)
+open import Data.List.Relation.Unary.All using (All)
+open import Data.Maybe using (Maybe)
+open import Data.Nat using (ℕ)
+open import Function using (_∘_)
+open import Level using (Level)
+
+open Maybe; open List; open ℕ
+
+private variable
+  ℓ : Level
+  A B : Set ℓ
 
 -- Looking up an index into the list; fails when out-of-bounds.
-_⁉_ : ∀ {a}{A : Set a} → List A → Nat → Maybe A
+_⁉_ : ∀ {a}{A : Set a} → List A → ℕ → Maybe A
 []       ⁉ _     = nothing
 (x ∷ _)  ⁉ zero  = just x
 (_ ∷ xs) ⁉ suc n = xs ⁉ n
+
+-- LIST OPERATIONS --
+flatMap : (A → List B) → List A → List B
+flatMap f [] = []
+flatMap f (x ∷ xs) = f x ++ flatMap f xs
+
+-- every element in the first list occurs in the second list
+_⊆_ : (l l' : List A) → Set _
```

Comment:
This is defined in `Data.List.Relation.Binary.Subset.Propositional` with a different but equivalent definition. If you use that, you may be able to use some theorems about it rather than proving them yourself.